server/datastore/mysql/migrations/tables/20230419082126_AddMDMAppleSetupAssistants.go

```diff
@@ -0,0 +1,41 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230419082126, Down_20230419082126)
+}
+
+func Up_20230419082126(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+-- macos setup assistant stores at most 1 profile per team/no team
+CREATE TABLE mdm_apple_setup_assistants (
+    id      INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    -- team_id NULL is for no team (cannot use 0 with foreign key)
+    team_id INT(10) UNSIGNED NULL,
+    -- this field is 0 for global, and the team_id otherwise, and is
+    -- used for the unique index/constraint (team_id cannot be used
+    -- as it allows NULL).
+    global_or_team_id INT(10) UNSIGNED NOT NULL DEFAULT 0,
```

Comment:
Do you think we should use this pattern for other "no team" settings (like bootstrap package)?  In those other cases, I think we didn't include a foreign key constraint. IIRC the reason for not using the fk was so that new table could use zero in order to get the unique index/constraint for the new table. But  that approach misses out on the cascade  on delete (and I don't think that we've accounted for that with any separate cleanup).

Reply:
this is a great point, I don't think we're doing any cleanups for the other tables.

spitballing, but if we're going for a general solution, what about having an auxiliary table that does the mapping?

| team_id | global_or_team_id |
|---------|-------------------|
| NULL    | 0                 |
| 1       | 1                 |
| 2       | 2                 |

then `mdm_apple_setup_assistants`, `mdm_apple_bootstrap_packages` and others can have a FK + unique index set to `aux_table.global_or_team_id`.

Reply:
@roperzh I think your suggestion makes sense, though we should probably address that in a PR of its own (will need some migrations to populate the initial data and add the FK). Would be a bit nicer than this approach. One thing to note, though, is that the global_or_team_id field is strictly used in the datastore layer, it doesn't leak out to other layers, but it would still be nice to not need it at all.